File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
a = int(input())
count = 0

if a == 4 or a == 7:
    print("-1")
else:
    if a % 5 == 0:
        count = a // 5
    elif a % 5 == 1:
        count = ((a - (3 * 2)) // 5) + 2
    elif a % 5 == 2:
        count = ((a - (3 * 4)) // 5) + 4
    elif a % 5 == 3:
        count = (a - 3 * 1) // 5 + 1
    elif a % 5 == 4:
        count = ((a - 3 * 3) // 5) + 3
    print(count)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		int count = 0;
		
		int a = sc.nextInt();
		
		if(a == 4 || a == 7)
		{
			System.out.println("-1");
		}
		else
		{
			if(a%5 == 0)
			{
				count = a/5;
			}
			else if(a%5 == 1)
			{
				count = ((a-(3*2))/5) + 2;
			}
			else if(a%5 == 2)
			{
					count = ((a-(3*4))/5) + 4;
			}
			else if(a%5 == 3)
			{
				count = (a-3*1)/5 + 1;
			}
			else if(a%5 == 4)
			{
				count = ((a-3*3)/5) + 3;
			}
			System.out.println(count);
		}
	}

}
